lumberjack.go

```diff
@@ -356,6 +375,42 @@ func (l *Logger) oldLogFiles() ([]logInfo, error) {
 	return logFiles, nil
 }
 
+// compressLog compresses the log with given filename
+// using Gzip compression
+func compressLog(filename string) error {
+	// Open reader for backup logfile
+	reader, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+
```

Comment:
you can (and should) defer reader.Close() here even if you explicitly close it later... multiple closes are ok on files.